googet.go

```diff
@@ -161,8 +162,13 @@ func repoList(dir string) ([]string, error) {
 	if !allowUnsafeURL {
 		var srl []string
 		for _, r := range rl {
-			if strings.ToLower(r[0:5]) != "https" {
-				logger.Errorf("%s will not be used as a repository, only https endpoints will be used unless 'allowunsafeurl' is set to 'true' in googet.conf", r)
+			parsed, err := url.Parse(r)
+			if err != nil {
+				logger.Errorf("Failed to parse URL '%s', skipping repo", r)
+				continue
+			}
+			if parsed.Scheme != "https" && parsed.Scheme != "gs" {
```

Comment:
gs is a made up scheme by gsutil, the URL should instead be parsed to see it if matches a gcs path, see here for how this can be done:

Reply:
I'm missing the link/reference that "see here" refers to.

I also don't follow. This code does parse the URL and branch based on the scheme detected by golang's net/url library.

Do you mean we should validate that the bucket and object exist via a gcloud storage call and consider the repo insecure if it doesn't?

Reply:
of course I leave out the link:
https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/daisy/storage.go

The idea is the gs:// "scheme" is a  made up for use in gsutil, we should also be checking for storage.googleapis.com and other URLs that reference GCS locations.

Reply:
Those regexs should be cleaner than what you have with indexPath currently

Reply:
That makes sense now. 
Looks like daisy isn't exposed as a library, so copy-n-paste it is I guess. Fixes pending...

Reply:
The latest commit parses the URL for various compatible formats per request. I added a unit test to make sure the regexes worked as I expected. I also made the regexes parse out trailing slashes. I suspect that's technically incorrect, since gcs sort of supports '/' as the last character of an object name, but it's more human friendly to parse it out, especially since the URL googet asks for is just the base and will have files under it.

For the record, looks like gsutil has a bug w.r.t. objects that have '/' as the last character of the name. You can create and ls them, but cat, cp and rm all fail with "No URLs matched".